
From: Ingo Molnar <mingo@elte.hu>

The patch below fixes smp_processor_id() warnings that are triggered by
numa_node_id().

All uses of numa_node_id() in mm/mempolicy.c seem to use it as a 'hint'
only, not as a correctness number.  Once a node is established, it's used
in a preemption-safe way.  So the simple fix is to disable the checking for
numa_node_id().  But additional review would be more than welcome, because
this patch turns off the preemption-checking of numa_node_id() permanently.
Tested on amd64.

Signed-off-by: Ingo Molnar <mingo@elte.hu>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/include/linux/mmzone.h |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

diff -puN include/linux/mmzone.h~fix-smp_processor_id-warning-in-numa_node_id include/linux/mmzone.h
--- 25/include/linux/mmzone.h~fix-smp_processor_id-warning-in-numa_node_id	Fri Oct  8 13:40:38 2004
+++ 25-akpm/include/linux/mmzone.h	Fri Oct  8 13:40:38 2004
@@ -370,7 +370,7 @@ int lower_zone_protection_sysctl_handler
 
 #include <linux/topology.h>
 /* Returns the number of the current Node. */
-#define numa_node_id()		(cpu_to_node(smp_processor_id()))
+#define numa_node_id()		(cpu_to_node(_smp_processor_id()))
 
 #ifndef CONFIG_DISCONTIGMEM
 
_
